src/terms/Hamiltonian.jl

```diff
@@ -31,43 +31,45 @@ end
 # As an optimization we special-case nonlocal operators to apply them
 # instead on the full block and benefit from BLAS3
 @views function LinearAlgebra.mul!(Hψ::AbstractArray, H::HamiltonianBlock, ψ::AbstractArray)
```

Comment:
I'd rather avoid the extra level of indentation here...

Reply:
We could make a function of the same signature with a sensible name and then just alias `mul!` to that.

Reply:
I would suggest calling the timed function `hamiltonian_mul` and then just alias the `LinearAlgebra.mul!` to that function in one line.